### DISCUSSION TITLE
How to add the builder block name in modal delete confirmation

### DISCUSSION THREAD

**Discussion by devhoussam1998** [filamentphp/filament#9863] [Help] [answered]
I want to add the builder block name in modal delete confirmation

**Expected result:**

<img width="450" alt="Screenshot 2023-11-25 at 15 38 07" src="https://github.com/filamentphp/filament/assets/121255432/78ee5d0e-26fe-4422-9740-a2fc0722f5ee">

```
Builder::make('content')
    ->blocks([
        Builder\Block::make('heading')
            ->schema([
                TextInput::make('content'),
            ]),
    ])
    ->deleteAction(
        fn (Action $action) => $action->requiresConfirmation(),
    )
```

**Comment by FinnPaes** [comment, marked as answer]
Probably too late, but incase someone else is looking for an answer:
```php
->deleteAction(fn (Action $action): Action => $action->label(fn (array $arguments, array $state) => 'Delete ' . $state[$arguments['item']]['type'])->requiresConfirmation())
```

If you need other data, check the `$state[$arguments['item']]` for more block data.

I dont know if it's the correct way, but it works fine.
